vllm-proxy/src/app/api/v1/openai.py

```diff
@@ -140,14 +145,17 @@ async def signature(request: Request, chat_id: str, signing_algo: str = None):
     if chat_id not in cache:
         return error("Chat id not found or expired", "chat_id_not_found")
 
-    if signing_algo is None:
-        signing_algo = quote.signing_method
-    elif signing_algo not in [ED25519, ECDSA]:
-        return error("Invalid signing algorithm. Must be 'ed25519' or 'ecdsa'", "invalid_signing_algo")
-
     # Retrieve the cached request and response
     chat_data = cache[chat_id]
-    signature = quote.sign(chat_data)
+    signature = None
+    signing_algo = ECDSA if signing_algo is None else signing_algo
+    if signing_algo == ECDSA:
+        signature = ecdsa_quote.sign(chat_data)
+    elif signing_algo == ED25519:
+        signature = ed25519_quote.sign(chat_data)
```

Comment:
So based on introducing `Signer`, here we use xxx_signer.sign(chat_data)` instead of quote